### PR TITLE
dev/core#88 Make sure financial_type_id is set when contribution is created

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
When a contribution is created via a contribution page (with a confirmation page) the parameter `financial_type_id` is not set if the contribution is not being created with a recurring contribution.

https://lab.civicrm.org/dev/core/issues/88

Before
----------------------------------------
`financial_type_id` is not set when a non-recurring contribution is submitted via a contribution page (with a confirmation page).  But it is set when a recurring contribution is being created at the same time.

After
----------------------------------------
`financial_type_id` is always set for contribution params when submitted via a contribution page (with a confirmation page).
This makes it more consistent for use in payment processor extensions etc.

### CODE REVIEW

**Comment by mattwire** [comment]
> @mattwire so we are winding up with contributions with financial_type_id is null? Is that a regression?

I think it depends on if the payment processor wants to do some extra work to identify what the financial_type_id is meant to be, so we don't necessarily end up with no financial_type.  I found this while testing smartdebit with non-recurring payments.

**Comment by eileenmcnaughton** [comment]
Hmm - can you explain further? Payment processor should set the payment_instrument but that is separate

File: CRM/Contribute/Form/Contribution/Confirm.php [hunk 1]
```diff
@@ -740,13 +740,13 @@ public function postProcess() {
...
    * Comments from previous refactor indicate doubt as to what was going on.
    *
-   * @param int $contributionTypeId
+   * @param int $financialTypeId
    *
    * @return null|string
    */
-  public function wrangleFinancialTypeID($contributionTypeId) {
-    if (isset($paymentParams['financial_type'])) {
-      $contributionTypeId = $paymentParams['financial_type'];
```
Comment (monishdeb) [inline CRM/Contribute/Form/Contribution/Confirm.php]: This is a bit strange as ```$paymentParams``` is not a parameter nor initialised inside the function yet this is in existence, seems like something to do with past refactoring. Also we should change the variable name to ```financialTypeId```  from ```contributionTypeId``` as what we all know and in past why that column was renamed :) 

so chnages here would be:
```php
public function wrangleFinancialTypeID($financialTypeId) {
 if (empty($financialTypeID) && !empty($this->_values['pledge_id'])) {
   $contributionTypeId = CRM_Core_DAO::getFieldValue('CRM_Pledge_DAO_Pledge',
         $this->_values['pledge_id'],
         'financial_type_id'
       );
}
 return $financialTypeId;
}
```

Comment (mattwire) [reply CRM/Contribute/Form/Contribution/Confirm.php]: @monishdeb Thanks for the review.  I've updated that function to match what you suggested as that's much cleaner.

**Comment by monishdeb** [comment]
Jenkins test this please

**Comment by monishdeb** [comment]
* ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) __Pass__
* ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) __Pass__
* ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) __Pass__
* ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) __Pass__
* ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) __Pass__
* ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) __Pass__ Tested on local, works fine, doesn't cause any intended regression
* ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) __Pass__
* ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) __Pass__

**Comment by eileenmcnaughton** [comment]
@monishdeb @mattwire I closed the gitlab issue - as a matter of process everyone (submitted, merger, review etc) should feel it's incumbent on them to close the issue since 2/3 will probably forget :-)

**Comment by monishdeb** [comment]
Thanks @eileenmcnaughton My bad that I forgot to close the ticket :(

**Pull request merged**